### ISSUE TITLE
Python API

### ISSUE THREAD

**Issue by doyubkim** [doyubkim/fluid-engine-dev#13]
Python binding will improve the productivity. The binding solution is not determined, yet. The candidates are Boost.Python, SWIG, and pybind11. Interoperating with NumPy and Matplotlib would be ideal.

**Comment by danraeprayStudentAcct** [comment]
Hi,

Just ordered your book - looking forward to reading it and going through the code here!  How are the python bindings coming along?  Are these tested / ready to use?

Thanks,

**Comment by doyubkim** [comment]
Thanks! Hope you like the book :)

The Python binding is still WIP, but I'm hoping to have it soon.

**Comment by doyubkim** [comment]
Let me share some updates on this ticket.

I added a Python binding using `pybind11` in a separate branch [`python`](https://github.com/doyubkim/fluid-engine-dev/tree/python). This new Python API covers most of the high-level C++ API, and I tried to add as much as docstrings as possible. However, it still lacks tests and examples, and the API may change a little bit. But I hope these will be added quite soon. Once all the tests/docs/examples are in place, it will be merged to the master.

@spockStudentAcct, feel free to test drive the (beta) API if you want.
